### PR TITLE
feat: add upscale task

### PR DESCRIPTION
**Pull request by ZiniuYu**
This PR adds the support to the newly added upscale task in inference-client.

This PR also enforces named arguments

### CODE REVIEW

File: inference_client/tasks/generate.py [hunk 1]
```diff
@@ -17,29 +17,30 @@ class GenerationMixin:
...
 
-        :param prompts: the prompts to generate text from
-        :param kwargs: additional arguments to pass to the model
+        :param prompts: the prompts to generate text from.
+        :param kwargs: additional arguments to pass to the model.
         """
         ...
 
     @overload
-    def generate(self, prompt: str, inplace_images: List = [], **kwargs):
```
Comment (ZiniuYu) [inline inference_client/tasks/generate.py:30]: A assume this is a typo

File: inference_client/tasks/upscale.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    token: str
+    client: Client
+
+    @overload
+    def upscale(
+        self,
+        *,
+        image: Union[str, bytes, 'ArrayType'],
+        output_width: Optional[int],
+        output_height: Optional[int],
```
Comment (numb3r3) [inline inference_client/tasks/upscale.py:27]:
```suggestion
        scale: Optional[str],
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: inference_client/tasks/upscale.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+    token: str
+    client: Client
+
+    @overload
+    def upscale(
+        self,
+        *,
+        image: Union[str, bytes, 'ArrayType'],
+        scale: Optional[str],
```
Comment (numb3r3) [inline inference_client/tasks/upscale.py:26]:
```suggestion
        scale: Optional[str] = None,
```

File: inference_client/tasks/upscale.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        **kwargs,
+    ):
+        """
+        Upscale plain input images. The result will be an image bytes.
```
Comment (numb3r3) [inline inference_client/tasks/upscale.py:30]: > The result will be an image bytes.

It is confusing. What's the expected image format? JPEG or PNG?

File: inference_client/tasks/upscale.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+        :param image: the image to upscale, can be a `ndarray`, 'bytes' or uri of the image
```
Comment (numb3r3) [inline inference_client/tasks/upscale.py:32]: if the `image` param is binary bytes, how to pass the image format?

**Review by numb3r3 — CHANGES_REQUESTED** [review]

**Comment by numb3r3** [comment]
FYI, maybe we can use this module to detect file type https://docs.python.org/3/library/imghdr.html#module-imghdr in our implementation

File: inference_client/tasks/caption.py [hunk 6]
```diff
@@ -19,7 +19,7 @@ class CaptionMixin:
     client: Client
 
     @overload
-    def caption(self, image: Union[str, bytes, 'ArrayType'], **kwargs):
+    def caption(self, *, image: Union[str, bytes, 'ArrayType'], **kwargs):
```
Comment (numb3r3) [inline inference_client/tasks/caption.py:22]: revert this change, I like the usage of:
```
t  = model.caption('/path/to/image.jpg')
```

File: inference_client/tasks/generate.py [hunk 7]
```diff
@@ -17,29 +17,30 @@ class GenerationMixin:
     client: 'Client'
 
     @overload
-    def generate(self, prompts: Union[str, List[str]], **kwargs):
+    def generate(self, *, prompts: Union[str, List[str]], **kwargs):
```
Comment (numb3r3) [inline inference_client/tasks/generate.py:20]: revert

File: inference_client/tasks/upscale.py [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+        ...
+
+    @overload
+    def upscale(
+        self,
+        *,
+        docs: Optional[Union[Iterable['Document'], 'DocumentArray']] = None,
+        image: Optional[Union[str, bytes, 'ArrayType']] = None,
+        scale: Optional[str] = None,
+        output_format: str = 'jpeg',
```
Comment (numb3r3) [inline inference_client/tasks/upscale.py:93]:
```suggestion
        image_format: str = 'jpeg',
```

as same as https://docarray.jina.ai/api/docarray.document.mixins.image/#docarray.document.mixins.image.ImageDataMixin.convert_image_tensor_to_blob

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: inference_client/tasks/upscale.py [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+    token: str
+    client: Client
+
+    @overload
+    def upscale(
+        self,
+        *,
+        image: Union[str, bytes, 'ArrayType'],
+        scale: Optional[str] = None,
+        output_format: str = 'jpeg',
```
Comment (numb3r3) [inline inference_client/tasks/upscale.py:28]: What's more, the format should also be optional.

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: inference_client/tasks/caption.py (hunk 6 above)
Comment (ZiniuYu) [reply inference_client/tasks/caption.py:22]: Ok. I would like to keep it for now since it is not simple as removing an asterisk. I will raise a separate PR for it.

**Review by numb3r3 — APPROVED** [review]
LGTM
